src/libraries/Microsoft.Extensions.Configuration.FileExtensions/ref/Microsoft.Extensions.Configuration.FileExtensions.cs

```diff
@@ -8,8 +8,8 @@ namespace Microsoft.Extensions.Configuration
 {
     public static partial class FileConfigurationExtensions
     {
-        public static System.Action<Microsoft.Extensions.Configuration.FileLoadExceptionContext> GetFileLoadExceptionHandler(this Microsoft.Extensions.Configuration.IConfigurationBuilder builder) { throw null; }
-        public static Microsoft.Extensions.FileProviders.IFileProvider GetFileProvider(this Microsoft.Extensions.Configuration.IConfigurationBuilder builder) { throw null; }
+        public static System.Action<Microsoft.Extensions.Configuration.FileLoadExceptionContext>? GetFileLoadExceptionHandler(this Microsoft.Extensions.Configuration.IConfigurationBuilder builder) { throw null; }
+        public static Microsoft.Extensions.FileProviders.IFileProvider? GetFileProvider(this Microsoft.Extensions.Configuration.IConfigurationBuilder builder) { throw null; }
```

Comment:
After the discussion at https://github.com/dotnet/runtime/pull/57434/files/d85eaa30ea23e2d89b8eeb1d1f05a4808dddd3b7#r780441763, I think this probably shouldn't be nullable. This should probably just throw if the `object` stored with the `FileProviderKey` is not an `IFileProvider`.